From 6ae3a5fffda83c3db34779e68c60322eecb65dd1 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 4 Jan 2024 12:20:20 +0100
Subject: [PATCH 0367/1695] dts: stm32: add tamper configuration on
 stm32mp135f-dk

Add the internal and external tamper configuration on stm32mp135f-dk
board.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 612f46a44a71bfef35e30f0f4537b015a5936996 dts: stm32: add tamper configuration on stm32mp135f-dk]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I7ecd717806f38f33615f0b9e145dbe603c7359d7
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/343135
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/349414
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp135f-dk.dts | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp135f-dk.dts b/core/arch/arm/dts/stm32mp135f-dk.dts
index cca157ddc..18a60fde0 100644
--- a/core/arch/arm/dts/stm32mp135f-dk.dts
+++ b/core/arch/arm/dts/stm32mp135f-dk.dts
@@ -11,6 +11,7 @@
 #include <dt-bindings/gpio/stm32mp_gpio.h>
 #include <dt-bindings/soc/stm32mp13-etzpc.h>
 #include <dt-bindings/regulator/st,stm32mp13-regulator.h>
+#include <dt-bindings/tamper/st,stm32mp13-tamp.h>
 #include "stm32mp135.dtsi"
 #include "stm32mp13xf.dtsi"
 #include "stm32mp13-pinctrl.dtsi"
@@ -517,6 +518,26 @@
 	vddsd2-supply = <&vdd>;
 };
 
+&tamp {
+	status = "okay";
+	st,tamp-passive-nb-sample = <4>;
+	st,tamp-passive-sample-clk-div = <16384>;
+
+	/* Tamper button */
+	tamp_button: tamp-passive@2 {
+		status = "okay";
+		reg = <EXT_TAMPER_2 0>;
+		pinctrl-0 = <&tamp0_in2_pin_a>;
+		st,tamp-mode = <TAMPER_CONFIRMED_MODE>;
+	};
+
+	/* Connect pin8 and pin22 from CN8 */
+	tamp-active@1 {
+		pinctrl-0 = <&tamp0_in3_pin_a &tamp0_out1_pin_a>;
+		status = "disabled";
+	};
+};
+
 &uart4 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart4_pins_a>;
-- 
2.47.2

